python/plugins/processing/algs/qgis/ExecuteSQL.py

```diff
@@ -112,21 +133,30 @@ def processAlgorithm(self, parameters, context, feedback):
         geometry_type = self.parameterAsEnum(parameters, self.INPUT_GEOMETRY_TYPE, context)
         geometry_crs = self.parameterAsCrs(parameters, self.INPUT_GEOMETRY_CRS, context)
 
+        scope = None
+        if self.INPUT_PARAMETERS in parameters:
+            scope = parameters[self.INPUT_PARAMETERS]
+
         df = QgsVirtualLayerDefinition()
         for layerIdx, layer in enumerate(layers):
             df.addSource('input{}'.format(layerIdx + 1), layer.id())
 
         if query == '':
             raise QgsProcessingException(
                 self.tr('Empty SQL. Please enter valid SQL expression and try again.'))
+        elif scope:
+            localContext = QgsExpressionContext()
```

Comment:
Here you should use the full processing algorithm expression context as a start:

    localContext  = self.createExpressionContext( parameters, context )
 
Then the expressions can utilise the global/project/algorithm scopes too.